File: Solutions/Python/Kth-smallest-element.py
#Problem link:https://practice.geeksforgeeks.org/problems/kth-smallest-element5635/1
 
def kthSmallest(arr, N, K):
    arr.sort()
    return arr[K-1]

# Driver code
if __name__ == '__main__':
    arr = [12, 3, 5, 7, 19]
    N = len(arr)
    K = 2
 
    # Function call
    print("K'th smallest element is",
          kthSmallest(arr, N, K))

File: Solutions/C++/CPP/Kthsmallestelement.cpp
// https://practice.geeksforgeeks.org/problems/kth-smallest-element5635/1
class Solution{
    public:
    // arr : given array
    // l : starting index of the array i.e 0
    // r : ending index of the array i.e size-1
    // k : find kth smallest element and return using this function
    int kthSmallest(int arr[], int l, int r, int k) {
        //code here
        // sort(arr,arr+r+1);
        priority_queue<int> pq;
        for(int i=0;i<r+1;i++)
        {        pq.push(arr[i]);
         if(pq.size()>k)
         pq.pop();
            }        
        return pq.top();
    }
};
